network-api/networkapi/wagtailpages/tests/buyersguide/test_editorial_content_index.py

```diff
@@ -14,13 +25,55 @@ class BuyersGuideEditorialContentIndexPageTest(test_base.WagtailpagesTestCase):
     @classmethod
     def setUpTestData(cls):
         super().setUpTestData()
+
         cls.pni_homepage = buyersguide_factories.BuyersGuidePageFactory(
             parent=cls.homepage,
         )
         cls.content_index = buyersguide_factories.BuyersGuideEditorialContentIndexPageFactory(
             parent=cls.pni_homepage,
         )
 
+        cls.request_factory = test.RequestFactory()
+
+    def create_request(self, data: Optional[dict] = None) -> 'wsgi.WSGIRequest':
+        return self.request_factory.get(path=self.content_index.url, data=data)
+
+    def create_days_old_article(self, days: int):
+        return buyersguide_factories.BuyersGuideArticlePageFactory(
+            parent=self.content_index,
+            first_published_at=timezone.now() - datetime.timedelta(days=days),
+        )
+
+    @contextlib.contextmanager
+    def setup_content_index_with_pages_of_children(self):
+        """
+        Context manager setting up the content index with child pages.
+
+        This is implemented as a context manager to allow us to mock / override the
+        `items_per_page` attribute on the content index regardless of how often the
+        page instance is created. This is important for integration tests with the
+        Django test client, because the instance of the content index you set up with
+        the factory is not the same that the client is hitting. A new instance is
+        instantiated with the data from the datbase. Since `items_per_page` is not
+        saved in the database it would be different on that instance. Mocking allows
+        us to set a value we want to use for the test.
+
+        Use this is in a with statement. The generated child pages are bound to the
```

Comment:
```suggestion
        This is done in a with statement. The generated child pages are bound to the
```

Reply:
Also, excellent docs :ok_hand:

Reply:
I have tried to clarify this usage description a little more, because I think your change referred to the nested `with` statement that applies the mocking, which mine was meant as a description on how to use the method.

Please let me know if I am off here.